task_sdk/src/airflow/sdk/log.py

```diff
@@ -0,0 +1,377 @@
+
+            if group:
+                # Only remove it from event_dict if we handle it
+                del event_dict["exc_info"]
+                event_dict["exception"] = list(
+                    itertools.chain.from_iterable(
+                        format_exception((type(exc), exc, exc.__traceback__))  # type: ignore[attr-defined,arg-type]
+                        for exc in (*group.exceptions, group)
+                    )
+                )
+
+        return event_dict
+
+    return _exception_group_tracebacks
+
+
+def logger_name(logger: Any, method_name: Any, event_dict: EventDict) -> EventDict:
+    if logger_name := event_dict.pop("logger_name", None):
+        event_dict.setdefault("logger", logger_name)
+    return event_dict
+
+
+def redact_jwt(logger: Any, method_name: str, event_dict: EventDict) -> EventDict:
+    for k, v in event_dict.items():
+        if isinstance(v, str) and v.startswith("eyJ"):
+            event_dict[k] = "eyJ***"
+    return event_dict
+
+
+def drop_positional_args(logger: Any, method_name: Any, event_dict: EventDict) -> EventDict:
+    event_dict.pop("positional_args", None)
+    return event_dict
+
+
+def json_processor(logger: Any, method_name: Any, event_dict: EventDict) -> str:
+    """Encode event into JSON format."""
+    return msgspec.json.encode(event_dict).decode("ascii")
+
+
+class StdBinaryStreamHandler(logging.StreamHandler):
+    """A logging.StreamHandler that sends logs as binary JSON over the given stream."""
+
+    stream: BinaryIO
+
+    def __init__(self, stream: BinaryIO):
+        super().__init__(stream)
+
+    def emit(self, record: logging.LogRecord):
+        try:
+            msg = self.format(record)
+            buffer = bytearray(msg, "ascii", "backslashreplace")
+
+            buffer += b"\n"
+
+            stream = self.stream
+            stream.write(buffer)
+            self.flush()
+        except RecursionError:  # See issue 36272
+            raise
+        except Exception:
+            self.handleError(record)
+
+
+@cache
+def logging_processors(
+    enable_pretty_log: bool,
+):
+    if enable_pretty_log:
+        timestamper = structlog.processors.MaybeTimeStamper(fmt="%Y-%m-%d %H:%M:%S.%f")
+    else:
+        timestamper = structlog.processors.MaybeTimeStamper(fmt="iso")
+
+    processors: list[structlog.typing.Processor] = [
+        timestamper,
+        structlog.contextvars.merge_contextvars,
+        structlog.processors.add_log_level,
+        structlog.stdlib.PositionalArgumentsFormatter(),
+        logger_name,
+        redact_jwt,
+        structlog.processors.StackInfoRenderer(),
+    ]
+
+    if enable_pretty_log:
+        # Imports to suppress showing code from these modules
+        import asyncio
+        import contextlib
+
+        import click
+        import httpcore
+        import httpx
+        import typer
+
+        rich_exc_formatter = structlog.dev.RichTracebackFormatter(
+            extra_lines=0,
+            max_frames=30,
```

Comment:
Any reason for 0 and 30 here than the defaults?

Reply:
Likely not! Picked largely at random. `extra_lines=0` was to make the stack traces smaller (as otherwise they get huge fast). max frames was an arbitrary choice

Reply:
Left a comment saying as much